src/riak_ensemble_peer.erl

```diff
@@ -1802,6 +1844,18 @@ setup({init, Args}, State0=#state{id=Id, ensemble=Ensemble, ets=ETS, mod=Mod}) -
     probe(init, State3).
 
 -spec handle_event(_, atom(), state()) -> {next_state, atom(), state()}.
+handle_event({watch_leader_status, Pid}, StateName, State) when node(Pid) =/= node() ->
+    lager:warning("Remote pid ~p not allowed to watch_leader_status on ensemble peer ~p",
+                  [Pid, State#state.id]),
+    {next_state, StateName, State};
+handle_event({watch_leader_status, Pid}, StateName, State = #state{watchers = Watchers}) ->
+    _ = notify_leader_status(Pid, StateName, State),
+    %% Might as well take this opportunity to prune any dead pids that are in the list
+    NewWatcherList = [P || P <- [Pid | Watchers], is_process_alive(P)],
```

Comment:
Yea - post-merge let's discuss splitting the 'DOWN' handling into a case where the Pid is a "watcher" and the default case, and then remove this stuff... just doesn't feel right the way it is. As I said, the other option would to be to prune the list every time we use it (start_watching, stop_watching, and any time we do a notification) so at least we'd be cleaning it up more regularly, but I'd rather just monitor/handle the 'DOWN' message correctly.